src/jupyter_contrib_nbextensions/nbextensions/code_prettify/code_prettify.js

```diff
@@ -157,52 +157,42 @@ define(function(require, exports, module) {
         };
     }
 
+    function getKernelInfos() {
+        //console.log("--->kernel_ready.Kernel")
+        kName = Jupyter.notebook.kernel.name;
+        console.log("kName",kName)
+        // get the list of user kernels and extract the language tag
+        $.getJSON(Jupyter.notebook.kernel.ws_url.replace("ws:", "http:") + '/api/kernelspecs',
```

Comment:
There is also `/api/kernelspecs/<name>` https://github.com/jupyter/notebook/blob/master/notebook/services/kernelspecs/handlers.py#L86

What I don't understand is why this information isn't available already? the editor component knows what kind of highlighting it should do, so that information should be available somewhere in the UI...